lib/doorkeeper/models/concerns/revocable.rb

```diff
@@ -8,6 +8,29 @@ def revoke(clock = Time)
       def revoked?
         !!(revoked_at && revoked_at <= Time.now)
       end
+
+      def revoke_in(time)
+        update_attribute :revoked_at, Time.now + time
+      end
```

Comment:
@tute tried your suggestion here `update(revoked_at: Time.now + time)` but it was the once causing the failed tests on 3.2